core/app/models/spree/store_credit.rb

```diff
@@ -23,10 +23,12 @@ class Spree::StoreCredit < Spree::PaymentSource
   validates_numericality_of :amount_used, { greater_than_or_equal_to: 0 }
   validate :amount_used_less_than_or_equal_to_amount
   validate :amount_authorized_less_than_or_equal_to_amount
+  validates :expires_at, presence: true, unless: ->(credit) { credit.category.non_expiring? }
```

Comment:
Sorry, I don't follow why it should only be allowed on a `non_expiring` category?

My thinking was that if we have an `expiring` category, one would expect those credits to have an expiration date (otherwise they wouldn't be expiring). Similarly if a credit is `non_expiring` it should not have an expiration date. If anything I would consider adding a validation that a `non_expiring` credit does not have an expiration date set.